Gameplay interactions

Frickerson opened the pull request.

Commit:
Create interaction classes

Commit:
Rename comment

Commit:
Fixup trace channel

Init members in cpp instead of header.

Commit:
Remove at

Commit:
Update interact GA

Commit:
Create interactable actor class

Commit:
Create interact GA

Commit:
Merge remote-tracking branch 'origin/develop' into gameplay_interactions

Commit:
Fix nullchecks + update options correctly

Commit:
Give ability if not present yet

Commit:
Add option to aim without player controller.

TheEmidee commented on Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp:
```diff
@@ -0,0 +1,117 @@
+#include "Interaction/Abilities/GBFGameplayAbility_Interact.h"
+
+#include "AbilitySystemComponent.h"
```
<>

Frickerson replied:
Done.

TheEmidee commented on Source/GameBaseFramework/Private/Interaction/Abilities/GBFGameplayAbility_Interact.cpp:
```diff
@@ -0,0 +1,117 @@
+#include "Interaction/GBFInteractionOption.h"
+#include "Interaction/GBFInteractionStatics.h"
+#include "Interaction/Tasks/GBFAT_WaitForInteractableTargets.h"
+#include "NativeGameplayTags.h"
```
<>

Frickerson replied:
Done.

TheEmidee commented on Content/Interaction/GA_Interact.uasset:
Should end with _P

Frickerson replied:
Done.

TheEmidee commented on Source/GameBaseFramework/Public/Interaction/Abilities/GBFGameplayAbilityTargetActor_Interact.h:
```diff
@@ -0,0 +1,17 @@
+
+/** Intermediate base class for all interaction target actors. */
+UCLASS( Blueprintable )
+class GAMEBASEFRAMEWORK_API AGBFGameplayAbilityTargetActor_Interact : public AGameplayAbilityTargetActor_Trace
```
I would inherit from `AGASExtGameplayAbilityTargetActor`

Frickerson replied:
It currently inherits from `AGameplayAbilityTargetActor_Trace` but the GASExt class inherits from `AGameplayAbilityTargetActor`. So some functionality is not there in that class which is used here.

TheEmidee commented on Source/GameBaseFramework/Public/Interaction/GBFInteractionStatics.h:
```diff
@@ -0,0 +1,27 @@
+struct FOverlapResult;
+
+UCLASS()
+class GAMEBASEFRAMEWORK_API UGBFInteractionStatics : public UBlueprintFunctionLibrary
```
final

Frickerson replied:
Done

TheEmidee commented on Source/GameBaseFramework/Public/Interaction/Tasks/GBFAT_WaitForInteractableTargets.h:
```diff
@@ -0,0 +1,41 @@
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam( FGBFInteractableObjectsChangedEvent, const TArray< FGBFInteractionOption > &, interactable_options );
+
+UCLASS( Abstract )
+class GAMEBASEFRAMEWORK_API UGBFAT_WaitForInteractableTargets : public UAbilityTask
```
could it be possible to inherit from `UGASExtAT_WaitTargetDataHitScan` or another GASExt class?

Frickerson replied:
not sure, since those tasks all end after they send the data, and this task keeps running.
We would have to override certain functions and change a bit how this task works, but I don't see much we would gain from doing that.

Commit:
Merge branch 'develop' into gameplay_interactions

Commit:
PR fixes

CS.

Commit:
Remove target actor class

Commit:
Use target data to update interactions

Fix nullcheck not checking if interface is valid.
Move code to interact ability.
Add static function to get interactable targets from target data.
Test in GA.

Commit:
Fix trace

Commit:
Inherit from GASExt hitscan AT

Setup looping the AT.

Commit:
Update GA to use the looping task

Commit:
Remove unnecessary class

Remove unused includes.

Commit:
Merge branch 'develop' into gameplay_interactions
